docs/m3db/architecture/engine.md

```diff
@@ -130,75 +109,138 @@ A read begins when an M3DB client calls the [`FetchBatchResult`](https://github.
 2. The series ID (byte blob)
 3. The period of time being requested (start and end)
 
-M3DB will consult the database object to check if the namespace exists, and if it does, then it will hash the series ID to determine which shard it belongs to. If the node receiving the read owns that shard, then M3DB needs to determine two things:
+M3DB will consult the database object to check if the namespace exists, and if it does, it will hash the series ID to determine which shard it belongs to. If the node receiving the read owns that shard, then M3DB needs to determine two things:
 
-1. Does the series exist? and if it does
-2. Does the data exist in an "active buffer" (actively being compressed by an encoder), cached in-memory, on disk, or some combination of all three?
+1. Whether the series exists and if it does,
+2. Whether the data exists in the buffer, cached in-memory, on disk, or some combination of all three.
 
 Determining whether the series exists is simple. M3DB looks up the series in the shard object. If it exists, then the series exists. If it doesn't, then M3DB consults an in-memory bloom filter(s) for that shard / block start combination(s) to determine if the series exists on disk.
 
-If the series exists, then for every block that the request spans, M3DB needs to consolidate data from the active buffers, in-memory cache, and fileset files (disk).
+If the series exists, then for every block that the request spans, M3DB needs to consolidate data from the buffer, in-memory cache, and fileset files (disk).
 
-Lets imagine a read for a given series that requests the last 6 hours worth of data, and an M3DB namespace that is configured with a blocksize of 2 hours (i.e we need to find 3 different blocks.)
+Let's imagine a read for a given series that requests the last 6 hours worth of data, and an M3DB namespace that is configured with a block size of 2 hours, i.e. we need to find 3 different blocks.
 
 If the current time is 8PM, then the location of the requested blocks might be as follows:
 
 ```
-[2PM - 4PM (FileSet file)]    - Sealed and flushed block that isn't cached
-[4PM - 6PM (In-memory cache)] - Sealed and flush block that is cached
-[6PM - 8PM (active buffer)]   - Hasn't been sealed or flushed yet
+[2PM - 4PM (FileSet file)]    - Flushed block that isn't cached
```

Comment:
Is it worth expanding this example with an unflushed coldwrite (for the 2PM-4PM) block?